src/event.rs

```diff
@@ -0,0 +1,587 @@
+			// TODO OnChainPaymentSent
+			//}
+			//6u8 => {
+			// TODO OnChainPaymentReceived
+			//}
+			_ => Err(lightning::ln::msgs::DecodeError::InvalidValue),
+		}
+	}
+}
+
+impl Writeable for LdkLiteEvent {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), lightning::io::Error> {
+		match self {
+			Self::PaymentSuccessful(event) => write_event(event, writer)?,
+			Self::PaymentFailed(event) => write_event(event, writer)?,
+			Self::PaymentReceived(event) => write_event(event, writer)?,
+			Self::ChannelClosed(event) => write_event(event, writer)?,
+		}
+		Ok(())
+	}
+}
+
+/// A payment we sent was successful.
+#[derive(Debug, Clone)]
+pub struct PaymentSuccessfulEvent {
+	/// The hash of the payment.
+	pub payment_hash: PaymentHash,
+}
+
+impl EventType for PaymentSuccessfulEvent {
+	const TYPE: u8 = 0u8;
+}
+
+impl Readable for PaymentSuccessfulEvent {
+	fn read<R: lightning::io::Read>(
+		reader: &mut R,
+	) -> Result<Self, lightning::ln::msgs::DecodeError> {
+		let payment_hash: PaymentHash = Readable::read(reader)?;
+		Ok(Self { payment_hash })
+	}
+}
+
+impl Writeable for PaymentSuccessfulEvent {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), lightning::io::Error> {
+		self.payment_hash.write(writer)?;
+		Ok(())
+	}
+}
+
+/// A payment we sent has failed.
+#[derive(Debug, Clone)]
+pub struct PaymentFailedEvent {
+	/// The hash of the payment.
+	pub payment_hash: PaymentHash,
+}
+
+impl EventType for PaymentFailedEvent {
+	const TYPE: u8 = 1u8;
+}
+
+impl Readable for PaymentFailedEvent {
+	fn read<R: lightning::io::Read>(
+		reader: &mut R,
+	) -> Result<Self, lightning::ln::msgs::DecodeError> {
+		let payment_hash: PaymentHash = Readable::read(reader)?;
+		Ok(Self { payment_hash })
+	}
+}
+
+impl Writeable for PaymentFailedEvent {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), lightning::io::Error> {
+		self.payment_hash.write(writer)?;
+		Ok(())
+	}
+}
+
+/// A payment has been received.
+#[derive(Debug, Clone)]
+pub struct PaymentReceivedEvent {
+	/// The hash of the payment.
+	pub payment_hash: PaymentHash,
+	/// The value, in thousandths of a satoshi that has been received.
+	pub amount_msat: u64,
+}
+
+impl EventType for PaymentReceivedEvent {
+	const TYPE: u8 = 2u8;
+}
+
+impl Readable for PaymentReceivedEvent {
+	fn read<R: lightning::io::Read>(
+		reader: &mut R,
+	) -> Result<Self, lightning::ln::msgs::DecodeError> {
+		let payment_hash: PaymentHash = Readable::read(reader)?;
+		let amount_msat: u64 = Readable::read(reader)?;
+		Ok(Self { payment_hash, amount_msat })
+	}
+}
+
+impl Writeable for PaymentReceivedEvent {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), lightning::io::Error> {
+		Self::TYPE.write(writer)?;
+		self.payment_hash.write(writer)?;
+		self.amount_msat.write(writer)?;
+		Ok(())
+	}
+}
+
+/// A channel has been closed.
+#[derive(Debug, Clone)]
+pub struct ChannelClosedEvent {
+	/// The channel_id of the channel which has been closed.
+	pub channel_id: [u8; 32],
```

Comment:
Mh, I'd actually love to use the `user_channel_id`, also since it would be simpler to work with in bindings. However, in its current form that would mean we receive a generic value of `0` if any inbound channel closes IIUC? So the event would say "one of your inbound channels closed, good luck finding out which". But, could we maybe start randomizing the default value for inbound channels? Then using `user_channel_id` as the one user-facing ID would indeed make much more sense.

Reply:
Now opened an upstream PR for this: https://github.com/lightningdevkit/rust-lightning/pull/1790